### ISSUE TITLE
Feature Request: Update daily sensor automatically when the actual sensor changes it's value

### ISSUE THREAD

**Issue by Vedeneb** [jeroenterheerdt/HADailySensor#5]
Hi,
would it be possible to have the daily sensor not to update after a specfifc time interval but instead whenever the main sensor changes it's value? My goal is to display the min and max temperature next to the actual temperature on my Dashboard and it looks kind of wrong if it get's warmer than the current max temp but the sensor takes a while to update and thus shows a wrong value. Sure, I could just reduce the time interval but imo the better alternative would be to automatically update the daily sensor based on state changes.

**Comment by Vedeneb** [comment]
The easier alternative would perhaps be to create an update service beside the reset service. That way I could just create an automation that calls the update service whenever the temperature changes

**Comment by jeroenterheerdt** [comment]
Right, I could add the update service, because the alternative cannot work - the component can not force itself to capture events that are not fired ;)

**Comment by jeroenterheerdt** [comment]
done. v0.3 provides this service.

**Comment by Vedeneb** [comment]
Wow, thanks a lot. Although I don't quite understand what you mean with "capture events that are not fired"? As soon as as any sensor changes it's value home assistant will send a state_changed event which could be captured by this integration or do I miss something?
Don't get me wrong - I'm absolutely happy with the new update service :) I'd just like to know why the other method can't work

**Comment by jeroenterheerdt** [comment]
yeah, you're right - I am just not a big fan of the way that works. So, sure, technically we could do this, but I don't think it is a good idea. Suppose people will add 500 of these Daily Sensors? Then HA gets to deal with 500 listeners to the update events, not sure if that is what we want to stimulate.

**Comment by erikgeurts** [comment]
A question to ensure I've understood this correctly:
- I've got a smart meter that reports the voltage on my grid connection in near real-time (every 2 to 3 seconds)
- I'd like to use this integration to keep a record of the minimum and maximum voltage over the course of a day, so that I can easily see if there have been any outliers below or above the officially allowed range of voltages
- low or high voltages can occur extremely briefly (few seconds), so polling the input sensor once a minute would likely miss these occurrences
- so instead I'd like this integration to automatically 'catch' any value that is lower than the previously reported minimum or higher than the previously reported maximum

Is this 'update service' the way to go to accomplish what I've described above?

If so, how do I configure this in the integration? Does anyone have practical experience with a similar scenario?
